WordPress/Classes/Services/PostHelper.m

```diff
@@ -238,7 +248,12 @@ + (NSArray *)mergePosts:(NSArray <RemotePost *> *)remotePosts
 {
     NSMutableArray *posts = [NSMutableArray arrayWithCapacity:remotePosts.count];
     for (RemotePost *remotePost in remotePosts) {
-        AbstractPost *post = [blog lookupPostWithID:remotePost.postID inContext:context];
+        AbstractPost *post;
+        if (remotePost.foreignID != nil) {
```

Comment:
(nit) might be worth storing the value as the metadata lookup is relatively expensive

Reply:
I'll address this in a future PR (targeting 24.9) since I'll have to update WPKit.